src/types/union.ts

```diff
@@ -993,6 +993,7 @@ export function Union(...alternatives: Rt[]): any {
               const field = alternative.reflect.fields[fieldName];
               if (
                 field.tag === 'literal' &&
+                value !== null &&
```

Comment:
This "fixes" it, but it's still not clear to me why the other test is not hitting  this.